go/libraries/doltcore/sqle/tables.go

```diff
@@ -568,8 +592,10 @@ func (t *AlterableDoltTable) CreateForeignKey(ctx *sql.Context, fkName string, c
 			Length: 0,
 		}
 		if !tblCols[i].TypeInfo.Equals(tableCol.TypeInfo) {
-			//TODO: make this more descriptive about our lack of support of different types for fks, figure it out later
-			return fmt.Errorf("column type mismatch on `%s` and `%s`", columns[i], tableCol.Name)
+			return fmt.Errorf("Column type mismatch on `%s` and `%s`. In other databases, we feel that type "+
```

Comment:
I wouldn't explain our thinking so thoroughly. Just note that the types of columns must match exactly.